### PR TITLE
Fix server uri storage to support having no workspace

### PR DESCRIPTION
**Pull request by rchiodo**
For #4037 

The JupyterServerUriStorage class would throw an exception while trying to save the 'serverType' setting if the user doesn't have a workspace open. This makes the enter key actually not close the quick pick dialog.

-   [x] Pull request represents a single change (i.e. not fixing disparate/unrelated things in a single PR).
-   [x] Title summarizes what is changing.
-   [x] Has a [news entry](https://github.com/Microsoft/vscode-jupyter/tree/main/news) file (remember to thank yourself!).
-   [x] Appropriate comments and documentation strings in the code.
-   [ ] Has sufficient logging.
-   [ ] Has telemetry for enhancements.
-   [x] Unit tests & system/integration tests are added/updated.
-   [ ] [Test plan](https://github.com/Microsoft/vscode-jupyter/blob/main/.github/test_plan.md) is updated as appropriate.
-   [ ] [`package-lock.json`](https://github.com/Microsoft/vscode-jupyter/blob/main/package-lock.json) has been regenerated by running `npm install` (if dependencies have changed).

### CODE REVIEW

File: src/client/datascience/jupyter/serverUriStorage.ts [hunk 1]
```diff
@@ -124,27 +124,25 @@ export class JupyterServerUriStorage implements IJupyterServerUriStorage {
...
 
             // Save in the storage (unique account per workspace)
             const key = this.getUriAccountKey();
             await this.encryptedStorage.store(Settings.JupyterServerRemoteLaunchService, key, uri);
         }
     }
 
+    private async updateServerType(val: string): Promise<void> {
+        if (this.workspaceService.hasWorkspaceFolders) {
+            return this.configService.updateSetting('jupyterServerType', val, undefined, ConfigurationTarget.Workspace);
```
Comment (DonJayamanne) [inline src/client/datascience/jupyter/serverUriStorage.ts:140]: As `jupyterServerType` is now configured to be `application`, the target needs to be global & not workspace.

File: src/client/datascience/jupyter/serverUriStorage.ts [hunk 2]
```diff
@@ -124,27 +124,25 @@ export class JupyterServerUriStorage implements IJupyterServerUriStorage {
...
+            await this.updateServerType(Settings.JupyterServerRemoteLaunch);
 
             // Save in the storage (unique account per workspace)
             const key = this.getUriAccountKey();
             await this.encryptedStorage.store(Settings.JupyterServerRemoteLaunchService, key, uri);
         }
     }
 
+    private async updateServerType(val: string): Promise<void> {
+        if (this.workspaceService.hasWorkspaceFolders) {
```
Comment (DonJayamanne) [inline src/client/datascience/jupyter/serverUriStorage.ts:139]: If you want this, then we need to revert the scope from `application` back to `resource`.

**Review by DonJayamanne — CHANGES_REQUESTED** [review]

File: src/client/datascience/jupyter/serverUriStorage.ts (hunk 1 above)
Comment (DonJayamanne) [reply src/client/datascience/jupyter/serverUriStorage.ts:140]: Please see next comment.

Comment (rchiodo) [reply src/client/datascience/jupyter/serverUriStorage.ts:140]: I guess I'm confused. I'll look it up in the VS code docs. But if the scope is global, does that mean it won't ever save it to the workspace settings?

Comment (rchiodo) [reply src/client/datascience/jupyter/serverUriStorage.ts:140]: Yes that's what VS code docs say:

```
application - Settings that apply to all instances of VS Code and can only be configured in user settings.
resource - Resource settings, which apply to files and folders, and can be configured in all settings levels, even folder settings.
```

**Review by DonJayamanne — APPROVED** [review]

**Pull request merged**